src/aznfswatchdog

```diff
@@ -113,32 +115,96 @@ reconcile_conntrack()
     done
 }
 
-vecho "Starting aznfswatchdog..."
 
-# Dump NAT table once on startup in case we have reported conflicts.
-vecho "NAT table:\n$(iptables-save -t nat)"
-conntrack -L > /dev/null
+#
+# Delete entry from $AZ_FILES_MOUNTMAP.
+#
+ensure_az_files_mountmap_not_exist()
+{
+    (
+        flock -e 999
 
-# conntrack timewait timeout higher than the TCP timewait timeout value isn't very valuable.
-conntrack_timeo_timew=$(cat /proc/sys/net/netfilter/nf_conntrack_tcp_timeout_time_wait 2>/dev/null)
-if [ $? -eq 0 -a $conntrack_timeo_timew -gt $AZNFS_TIMEWAIT_TIMEOUT ]; then
-        vecho "Changing /proc/sys/net/netfilter/nf_conntrack_tcp_timeout_time_wait [$conntrack_timeo_timew -> $AZNFS_TIMEWAIT_TIMEOUT]"
-        echo $AZNFS_TIMEWAIT_TIMEOUT > /proc/sys/net/netfilter/nf_conntrack_tcp_timeout_time_wait
-fi
+        #
+        # If user wants to delete the entry only if AZ_FILES_MOUNTMAP has not changed since
+        # he looked up, honour that.
+        #
+        local ifmatch="$2"
+        if [ -n "$ifmatch" ]; then
+            local mtime=$(stat -c%Y $AZ_FILES_MOUNTMAP)
+            if [ "$mtime" != "$ifmatch" ]; then
+                eecho "[$1] Refusing to remove from ${AZ_FILES_MOUNTMAP} as $mtime != $ifmatch!"
+                return 1
+            fi
+        fi
 
-if ! chattr -f +i $MOUNTMAP; then
-    wecho "chattr does not work for ${MOUNTMAP}!"
-fi
+	chattr -f -i $AZ_FILES_MOUNTMAP
```

Comment:
>chattr -f -i $AZ_FILES_MOUNTMAP [](http://example.com/codeflow?start=1&length=31)

why is only this command sticking out?

Reply:
chattr -f -i command is used to remove the immutable attribute from $AZ_FILES_MOUNTMAP file. So whenever we want to modify, delete $AZ_FILES_MOUNTMAP file, we first need to remove the immutable attribute. Then modify or delete $AZ_FILES_MOUNTMAP. And then set the immutable attribute again to prevent the file from being modified or deleted.